docs/design/021-component-upgrade-safety.md

```
@@ -0,0 +1,545 @@
+
+and is referenced from the transition record:
+
+```yaml
+    hooks:
+      - file: migrations/adopt-mirrored-crs.yaml
+        phase: pre-upgrade
+```
+
+The bundler emits it as a separate generated chart folder, ordered immediately before the component it serves:
+
+```
```

markdownlint-cli2 (0.23.2)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)